visserver/visserver.go

```diff
@@ -128,11 +129,19 @@ func (server *Server) ClientConnected(client *wsserver.Client) {
 }
 
 // ClientDisconnected disconnect client notification.
-func (server *Server) ClientDisconnected(client *wsserver.Client) {
+func (server *Server) ClientDisconnected(wsClient *wsserver.Client) {
 	server.Lock()
 	defer server.Unlock()
 
-	delete(server.clients, client)
+	client, ok := server.clients[wsClient]
+	if !ok {
+		log.Error("Disconnect unknown client")
+		return
+	}
+
+	_ = client.unsubscribeAll()
```

Comment:
maybe it will be better to add a log error here

Reply:
done